app/src/main/java/com/appunite/loudius/ui/reviewers/ReviewersViewModel.kt

```diff
@@ -70,73 +70,74 @@ class ReviewersViewModel @Inject constructor(
         fetchData()
     }
 
-    private fun getInitialValues(savedStateHandle: SavedStateHandle) = InitialValues(
-        checkNotNull(savedStateHandle[Screen.Reviewers.ownerArg]),
-        checkNotNull(savedStateHandle[Screen.Reviewers.repoArg]),
-        checkNotNull(savedStateHandle[Screen.Reviewers.pullRequestNumberArg]),
-        checkNotNull(savedStateHandle[Screen.Reviewers.submissionDateArg]),
-    )
-
     private fun fetchData() {
         viewModelScope.launch {
+            state = state.copy(isLoading = true, isError = false)
+
             getMergedData()
-                .onSuccess { state = state.copy(reviewers = it.orEmpty(), isLoading = false) }
+                .onSuccess { state = state.copy(reviewers = it, isLoading = false) }
                 .onFailure { state = state.copy(isError = true, isLoading = false) }
         }
     }
 
-    private suspend fun getMergedData(): Result<List<Reviewer>?> =
+    private suspend fun getMergedData(): Result<List<Reviewer>> = downloadData()
+        .mapData()
+
+    private suspend fun downloadData(): Pair<Result<RequestedReviewersResponse>, Result<List<Review>>> =
         coroutineScope {
-            state = state.copy(isLoading = true, isError = false)
-            val requestedReviewersDeferred = async { fetchRequestedReviewers() }
-            val reviewersDeferred = async { fetchReviews() }
+            val (owner, repo, pullRequestNumber) = initialValues
```

Comment:
suggestion: I wouldn't extract those parameters. It's a place for an issue, especially because those parameters have the same types.

Reply:
Sounds good for me - added 👌